runtime/compiler/net/ServerStream.cpp

```diff
@@ -0,0 +1,312 @@
+      exit(1);
+      }
+
+   // verify server identity using standard method
+   SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER, NULL);
+
+   if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
+      TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "Successfully initialized SSL context: OPENSSL_VERSION_NUMBER 0x%lx\n", OPENSSL_VERSION_NUMBER);
+
+   return ctx;
+   }
+
+static bool
+handleOpenSSLConnectionError(int connfd, SSL *&ssl, BIO *&bio, const char *errMsg)
+{
+   if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
+       TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "%s: errno=%d", errMsg, errno);
+   ERR_print_errors_fp(stderr);
+
+   close(connfd);
+   if (bio)
+      {
+      BIO_free_all(bio);
+      bio = NULL;
+      }
+   if (ssl)
+      {
+      SSL_free(ssl);
+      ssl = NULL;
+      }
+   return false;
+}
+
+static bool
+acceptOpenSSLConnection(SSL_CTX *sslCtx, int connfd, BIO *&bio)
+   {
+   SSL *ssl = SSL_new(sslCtx);
+   if (!ssl)
+      return handleOpenSSLConnectionError(connfd, ssl, bio, "Error creating SSL connection");
+
+   SSL_set_accept_state(ssl);
+
+   if (SSL_set_fd(ssl, connfd) != 1)
+      return handleOpenSSLConnectionError(connfd, ssl, bio, "Error setting SSL file descriptor");
+
+   if (SSL_accept(ssl) <= 0)
+      return handleOpenSSLConnectionError(connfd, ssl, bio, "Error accepting SSL connection");
+
+   bio = BIO_new_ssl(sslCtx, false);
+   if (!bio)
+      return handleOpenSSLConnectionError(connfd, ssl, bio, "Error creating new BIO");
+
+   if (BIO_set_ssl(bio, ssl, true) != 1)
+      return handleOpenSSLConnectionError(connfd, ssl, bio, "Error setting BIO SSL");
+
+   if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
+      TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "SSL connection on socket 0x%x, Version: %s, Cipher: %s\n",
+                                                     connfd, SSL_get_version(ssl), SSL_get_cipher(ssl));
+   return true;
+   }
+#endif
+
+void
+ServerStream::serveRemoteCompilationRequests(BaseCompileDispatcher *compiler, TR::PersistentInfo *info)
+   {
+#if defined(JITSERVER_ENABLE_SSL)
+   SSL_CTX *sslCtx = NULL;
+   if (CommunicationStream::useSSL())
+      {
+      CommunicationStream::initSSL();
+      sslCtx = createSSLContext(info);
+      }
+#endif
+
+   uint32_t port = info->getJITServerPort();
+   uint32_t timeoutMs = info->getSocketTimeout();
+   int sockfd = socket(AF_INET, SOCK_STREAM, 0);
+   if (sockfd < 0)
+      {
+      perror("can't open server socket");
+      exit(1);
+      }
+
+   // see `man 7 socket` for option explanations
+   int flag = true;
+   if (setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, (void *)&flag, sizeof(flag)) < 0)
+      {
+      perror("Can't set SO_REUSEADDR");
+      exit(-1);
+      }
+   if (setsockopt(sockfd, SOL_SOCKET, SO_KEEPALIVE, (void *)&flag, sizeof(flag)) < 0)
+      {
+      perror("Can't set SO_KEEPALIVE");
+      exit(-1);
+      }
+
+   struct timeval timeoutForListening = {0, 0}; // Wait forever
```

Comment:
The name doesn't match what it's used for (send/receive timeouts).

Reply:
Will be removed since by default the socket will wait forever.